app/controllers/api/databases.php

```diff
@@ -1753,15 +1752,14 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
         $indexes = $collection->getAttribute('indexes');
 
         // Search for index
-        $indexIndex = array_search($key, array_column($indexes, 'key'));
+        $indexIndex = array_search($key, array_map(fn($idx) => $idx['key'], $indexes));
 
         if ($indexIndex === false) {
             throw new Exception(Exception::INDEX_NOT_FOUND);
         }
 
-        $index = new Document([\array_merge($indexes[$indexIndex], [
-            'collectionId' => $database->getInternalId() . '_' . $collectionId,
-        ])]);
+        $index = $indexes[$indexIndex];
```

Comment:
Just seemed redundant to wrap the index in a new document instead of just setting the `collectionId`, but can be reverted once https://github.com/appwrite/appwrite/pull/3712 is merged